### PR TITLE
build(test-snapshots): use node16 module resolution

### PR DESCRIPTION
**Pull request by DLehenbauer**
Builds both ESM and CJS in test-snapshots.

Notable changes:

- CJS tests are skipped in CI.
- ts2esm updated all imports to use extensions.
- __dirname usage replaced by dirname.cts.
- I had to exclude the package from some policies around tsc and esnext scripts. I think these are false positives that Jason fixed in the latest build-tools, so I added exclusions.

### CODE REVIEW

File: packages/test/snapshots/package.json [hunk 1]
```diff
@@ -12,14 +12,14 @@
...
-	"main": "dist/generateSnapshotFiles.js",
-	"types": "dist/generateSnapshotFiles.js",
+	"type": "module",
 	"scripts": {
 		"build": "fluid-build . --task build",
 		"build:compile": "fluid-build . --task compile",
+		"build:esnext": "tsc --project ./tsconfig.json",
 		"build:genver": "gen-version",
-		"build:test": "tsc --project ./src/test/tsconfig.json",
-		"check:biome": "biome check .",
```
Comment (tylerbutler) [inline packages/test/snapshots/package.json:22]: Probably want to keep the check biome task

**Review by tylerbutler — CHANGES_REQUESTED** [review]
The biome task should stick around.

File: packages/test/snapshots/package.json [hunk 2]
```diff
@@ -31,23 +33,25 @@
...
+		"test:mocha": "npm run test:mocha:esm && echo skipping cjs to avoid overhead - npm run test:mocha:cjs",
+		"test:mocha:cjs": "mocha --ignore \"dist/test/generate/*\"  --recursive \"dist/test/**/*.spec.*js\" --exit",
+		"test:mocha:esm": "mocha --ignore \"lib/test/generate/*\" --recursive \"lib/test/**/*.spec.*js\" --exit",
 		"test:mocha:verbose": "cross-env FLUID_TEST_VERBOSE=1 npm run test:mocha",
-		"test:new": "mocha --experimental-worker \"dist/test/generate/new.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
-		"test:update": "mocha --experimental-worker \"dist/test/generate/update.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
-		"tsc": "tsc"
+		"test:new": "mocha --experimental-worker \"lib/test/generate/new.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
+		"test:update": "mocha --experimental-worker \"lib/test/generate/update.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
+		"tsc": "fluid-tsc commonjs --project ./tsconfig.cjs.json"
```
Comment (tylerbutler) [inline packages/test/snapshots/package.json:42]:
```suggestion
		"tsc": "fluid-tsc commonjs --project ./tsconfig.cjs.json && copyfiles -f ../../../common/build/build-common/src/cjs/package.json ./dist"
```

**Review by tylerbutler — APPROVED** [review]

File: packages/test/snapshots/package.json [hunk 3]
```diff
@@ -31,23 +33,25 @@
...
-		"test:mocha": "mocha --experimental-worker --ignore \"dist/test/generate/*\" dist/test",
+		"test:mocha": "npm run test:mocha:esm && echo skipping cjs to avoid overhead - npm run test:mocha:cjs",
+		"test:mocha:cjs": "mocha --ignore \"dist/test/generate/*\"  --recursive \"dist/test/**/*.spec.*js\" --exit",
+		"test:mocha:esm": "mocha --ignore \"lib/test/generate/*\" --recursive \"lib/test/**/*.spec.*js\" --exit",
 		"test:mocha:verbose": "cross-env FLUID_TEST_VERBOSE=1 npm run test:mocha",
-		"test:new": "mocha --experimental-worker \"dist/test/generate/new.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
-		"test:update": "mocha --experimental-worker \"dist/test/generate/update.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
-		"tsc": "tsc"
+		"test:new": "mocha --experimental-worker \"lib/test/generate/new.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
+		"test:update": "mocha --experimental-worker \"lib/test/generate/update.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
```
Comment (jason-ha) [inline packages/test/snapshots/package.json:41]: Why do these have `-r node_modules/@fluid-internal/mocha-test-setup` still here?

File: fluidBuild.config.cjs [hunk 4]
```diff
@@ -205,11 +205,13 @@ module.exports = {
 			],
 			"fluid-build-tasks-eslint": [
 				// eslint doesn't really depend on build. Doing so just slows down a package build.
+				"^packages/test/snapshots/package.json",
 				"^packages/test/test-utils/package.json",
 			],
 			"fluid-build-tasks-tsc": [
 				// TODO: AB#7460 fix tsconfig reference path match on Windows
 				"^packages/tools/devtools/devtools-view/package.json",
+				"^packages/test/snapshots/package.json",
```
Comment (jason-ha) [inline fluidBuild.config.cjs:214]: What is special about this package compared to others?
I know why tools/devtools-view has an exception, but doesn't seem like the referenced issue matches.

File: packages/test/snapshots/tsconfig.json [hunk 5]
```diff
@@ -1,14 +1,13 @@
...
 	"exclude": ["src/test/**/*"],
 	"compilerOptions": {
-		"strict": false,
 		"rootDir": "./src",
-		"outDir": "./dist",
+		"outDir": "./lib",
 		"types": ["node"],
+
+		// TODO: this should be updated to use strict mode ASAP
+		"strict": false,
```
Comment (jason-ha) [inline packages/test/snapshots/tsconfig.json:11]: Gotta work item?

File: fluidBuild.config.cjs (hunk 4 above)
Comment (tylerbutler) [reply fluidBuild.config.cjs:214]: I figured it out I think. The policy does not treat `tsc --project tsconfig.json` the same as `tsc --project ./tsconfig.json`. Will be fixed in upcoming PR.

**Pull request merged**